tests/TestCase/ORM/Behavior/TranslateBehaviorTest.php

```diff
@@ -1856,4 +1857,178 @@ public function testCustomTableLocator()
         $this->assertNotSame($table->associations()->getTableLocator(), $behaviorLocator);
         $this->assertNotSame($this->getTableLocator(), $behaviorLocator);
     }
+
+    /**
+     * Tests that using matching doesn't cause an association property to be created.
+     *
+     * @return void
+     */
+    public function testMatchingDoesNotCreateAssociationProperty()
+    {
+        $table = $this->getTableLocator()->get('Articles');
+        $table->hasMany('Comments');
+
+        $table->Comments->addBehavior('Translate');
+        $table->Comments->setLocale('abc');
+
+        $this->assertNotEquals($table->Comments->getLocale(), I18n::getLocale());
+
+        $result = $table
+            ->find()
+            ->matching('Comments')
+            ->first();
+
+        $this->assertArrayNotHasKey('comments', $result->toArray());
```

Comment:
This assertion is invalid. There are no `fields` specified when attaching the translate behavior so it essentially does nothing. 

If you specify `fields` the behavior will generate a contain internally and you will get `comments` property in result even though the "user" query only does a matching.